### DISCUSSION TITLE
outbound vmess协议address字段中域名的解析问题

### DISCUSSION THREAD

**Discussion by sk-buff** [v2fly/v2ray-core#536] [Q&A]
大家好，
我遇到了一个关于outbound vmess协议中address域名的解析问题。假设我的代理服务器的域名为example.com，该域名的对应IP为x.x.x.x，而该域名被填入了一个vmess协议outbound的vnext->address字段中。我创建了一个虚拟网卡并使用tun2socks实现全局代理，除了将tun的DNS设置为8.8.8.8外，我还将DHCP分配给我物理网卡的默认DNS（假设其为y.y.y.y）也改为了8.8.8.8，但是修改后我发现v2ray-core就无法将exmaple.com解析为IP地址了。这证明v2ray-core关于address字段中的域名解析默认是走系统代理的。为了让代理能够正常工作，我尝试不让example.com的解析走系统代理，但是我尝试在v2ray-core的config添加了“dns”:{"hosts": {"example.com": "x.x.x.x"}}或者“dns”:{"servers": [“y.y.y.y”]}都不能解决我的问题。最后还是通过修改了hosts文件才得以解决，但修改hosts还是意味着解析过程是由系统完成的。想请教一下是否有办法让example.com的解析由v2ray-core来进行而不是由系统进行？请大佬们不吝赐教，谢谢！

**Comment by kslr** [comment]
除非你把dns交给core来处理
